### PR TITLE
fixed last_play features and added three new features to f_player_season

### PR DESCRIPTION
**Pull request by dmf95**
# Overview
- Fixed the logic for the `last_play` as it was broken (see picture for solution in red)
- Added `penalty_shot_attempts`, `penalty_shot_goals`, and `empty_net_goals` to `f_player_season`

image

### CODE REVIEW

File: models/staging/stg_nhl__live_plays.sql [hunk 1]
```diff
@@ -212,7 +212,79 @@ live_plays as (
...
+                then lag(bp.event_idx, 1) over (partition by game_id order by bp.game_id, event_idx, play_time)
+            when lag(bp.event_idx, 2) over (partition by game_id order by bp.game_id, event_idx, player_index, play_time) != bp.event_idx
+                then lag(bp.event_idx, 2) over (partition by game_id order by bp.game_id, event_idx, play_time)
+            when lag(bp.event_idx, 3) over (partition by game_id order by bp.game_id, event_idx, player_index, play_time) != bp.event_idx
+                then lag(bp.event_idx, 3) over (partition by game_id order by bp.game_id, event_idx, play_time)
+            when lag(bp.event_idx, 4) over (partition by game_id order by bp.game_id, event_idx, player_index, play_time) != bp.event_idx
+                then lag(bp.event_idx, 4) over (partition by game_id order by bp.game_id, event_idx, play_time)
+            when lag(bp.event_idx, 5) over (partition by game_id order by bp.game_id, event_idx, player_index, play_time) != bp.event_idx
+                then lag(bp.event_idx, 5) over (partition by game_id order by bp.game_id, event_idx, play_time)
+        end as last_play_event_idx
```
Comment (gavh3) [inline models/staging/stg_nhl__live_plays.sql:227]: Can you help me follow what's happening here?

Comment (dmf95) [reply models/staging/stg_nhl__live_plays.sql:227]: oh, yea sure although it's incredibly confusing :)

so our original raw play-by-play data stored each play as an array... when we unlisted such that each player's involvement in the play had its own row, it unlisted in order of involvement... so like, goalscorer = row 1, primary assist = row 2, secondary assist row =3, etc... this logic here is basically to associate the last play with a player (see pic)

**Comment by dmf95** [comment]
[last_play_explanation.pptx](https://github.com/the-data-base/nhl-data/files/12449219/last_play_explanation.pptx)

Here is a study of the behaviour observed at the play level to explain how the new features should behave

**Review by gavh3 — DISMISSED** [review]

**Review by gavh3 — APPROVED** [review]
💸

**Pull request merged**